### PR TITLE
Added recalculation of CheckList for PunchListItem Creation, Deletion…

### PR DESCRIPTION
**Pull request by agustarnason**
… and certain Updates

### CODE REVIEW

**Comment by TordJoranger** [comment]
I'm not sure if this enpoint in webapi returns right away, and if it returns at all.  Either way, we should not await it/require it to work.

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/ClearPunchItem/ClearPunchItemCommandHandler.cs [hunk 1]
```diff
@@ -64,6 +68,8 @@ public async Task<Result<string>> Handle(ClearPunchItemCommand request, Cancella
 
             await _syncToPCS4Service.SyncPunchListItemUpdateAsync(integrationEvent, cancellationToken);
 
+            await _checkListApiService.RecalculateCheckListStatus(punchItem.Plant, punchItem.CheckListGuid, cancellationToken);
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/ClearPunchItem/ClearPunchItemCommandHandler.cs:71]: after commit imo

**Review by TordJoranger — CHANGES_REQUESTED** [review]
Fire and forget

**Review by janid1967 — CHANGES_REQUESTED** [review]
In xxxPunchItemCommandHandlerTests.cs, add a new test assering that _checkListApiService.RecalculateCheckListStatus is called with correct parameteres for plant and CheckListGuid

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/ClearPunchItem/ClearPunchItemCommandHandler.cs (hunk 1 above)
Comment (TordJoranger) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/ClearPunchItem/ClearPunchItemCommandHandler.cs:71]: yeah ok, disregard D:

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/Attachments/AttachmentServiceTests.cs
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.Completion.Command.Tests/Attachments/AttachmentServiceTests.cs:1]: remove unused usings

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/Links/LinkServiceTests.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@
 using Microsoft.Extensions.Logging;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using NSubstitute;
-using Equinor.ProCoSys.Completion.DbSyncToPCS4;
+using static Microsoft.Graph.Constants;
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.Completion.Command.Tests/Links/LinkServiceTests.cs:16]: remove (unused)

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/PunchItemCommands/ClearPunchItem/ClearPunchItemCommandHandlerTests.cs [hunk 3]
```diff
@@ -126,6 +127,28 @@ public async Task HandlingCommand_ShouldSendHistoryUpdatedIntegrationEvent()
     }
 
     #region Unit Tests which can be removed when no longer sync to pcs4
+    [TestMethod]
+    public async Task HandlingCommand_ShouldRecalculateChecklist()
+    {
+        // Arrange
+        PunchItemUpdatedIntegrationEvent integrationEvent = null!;
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.Completion.Command.Tests/PunchItemCommands/ClearPunchItem/ClearPunchItemCommandHandlerTests.cs:134]: remove thw whole Arrange block, integrationEvent not used

Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Completion.Command.Tests/PunchItemCommands/ClearPunchItem/ClearPunchItemCommandHandlerTests.cs:134]: same in other tests

**Review by janid1967 — APPROVED** [review]

**Comment by agustarnason** [comment]
> I'm not sure if this enpoint in webapi returns right away, and if it returns at all. Either way, we should not await it/require it to work.


@TordJoranger @janid1967  I think the question in regards to the recalculation being awaited on, is whether or not we should do a rollback if we fail to recalculate, as I do agree that we don't really need to wait for the recalculation to be done for us to finish
If we don't care about the recalculation endpoint failing for whatever reason or not, we can just fire and forget, otherwise we probably should keep the await

**Comment by TordJoranger** [comment]
> > I'm not sure if this enpoint in webapi returns right away, and if it returns at all. Either way, we should not await it/require it to work.
> 
> @TordJoranger @janid1967 I think the question in regards to the recalculation being awaited on, is whether or not we should do a rollback if we fail to recalculate, as I do agree that we don't really need to wait for the recalculation to be done for us to finish If we don't care about the recalculation endpoint failing for whatever reason or not, we can just fire and forget, otherwise we probably should keep the await

Ok, lets discuss it later, but we don't need to hold the pr over it.

**Review by TordJoranger — APPROVED** [review]

**Pull request merged**